src/utils/collection.rs

```diff
@@ -198,6 +198,42 @@ impl BtfCollection {
         Ok(ids)
     }
 
+    /// Find a list of BTF ids using a regex describing their name as a key.
+    /// Matching ids can be found in multiple underlying BTF, thus this function
+    /// returns a list of tuples containing each a reference to `NamedBtf`
+    /// (representing the BTF where a match was found) and the id. Further
+    /// lookups must be done using the `Btf` object contained in the linked
+    /// `NamedBtf` one.
+    #[cfg(feature = "regex")]
+    pub fn resolve_ids_by_regex(&self, re: &regex::Regex) -> Result<Vec<(&NamedBtf, u32)>> {
+        let mut ids = Vec::new();
+
+        let mut base_ids = match self.base.btf.resolve_ids_by_regex(re) {
+            Ok(base_ids) => base_ids,
+            _ => Vec::new(), // Id not found in base.
+        };
+
+        for split in self.split.iter() {
+            if let Ok(mut mod_ids) = split.btf.resolve_ids_by_regex(re) {
```

Comment:
Same comment as in the other PR. Could we also have a helper that does not search the base again?